autogpt_platform/autogpt_libs/autogpt_libs/utils/cache.py

```diff
@@ -200,48 +291,98 @@ def sync_wrapper(*args: P.args, **kwargs: P.kwargs):
 
+                # Compute redis_key once if using shared cache
+                redis_key = _make_redis_key(key)
+
                 # Fast path: check cache without lock
-                if key in cache_storage:
-                    if ttl_seconds is None:
-                        logger.debug(
-                            f"Cache hit for {target_func.__name__} args: {args} kwargs: {kwargs}"
+                if shared_cache:
+                    try:
+                        # Use GET directly instead of EXISTS + GET (saves 1 round trip)
+                        cached_bytes = redis_client.get(redis_key)
+                        if cached_bytes is not None and isinstance(cached_bytes, bytes):
+                            logger.debug(
+                                f"Cache hit for {target_func.__name__} args: {args} kwargs: {kwargs}"
+                            )
+                            return pickle.loads(cached_bytes)
+                    except Exception as e:
+                        logger.error(
+                            f"Redis error during cache check for {target_func.__name__}: {e}"
                         )
-                        return cache_storage[key]
-                    else:
-                        cached_data = cache_storage[key]
-                        if isinstance(cached_data, tuple):
-                            result, timestamp = cached_data
-                            if current_time - timestamp < ttl_seconds:
-                                logger.debug(
-                                    f"Cache hit for {target_func.__name__} args: {args} kwargs: {kwargs}"
-                                )
-                                return result
-
-                # Slow path: acquire lock for cache miss/expiry
-                with cache_lock:
-                    # Double-check: another thread might have populated cache
+                        # Fall through to execute function
+                        return target_func(*args, **kwargs)
+                else:
                     if key in cache_storage:
                         if ttl_seconds is None:
+                            logger.debug(
+                                f"Cache hit for {target_func.__name__} args: {args} kwargs: {kwargs}"
+                            )
                             return cache_storage[key]
                         else:
                             cached_data = cache_storage[key]
                             if isinstance(cached_data, tuple):
                                 result, timestamp = cached_data
                                 if current_time - timestamp < ttl_seconds:
+                                    logger.debug(
+                                        f"Cache hit for {target_func.__name__} args: {args} kwargs: {kwargs}"
+                                    )
                                     return result
 
+                # Slow path: acquire lock for cache miss/expiry
```

Comment:
Can we ask AI to simplify this? :D 
tell it to feel free refactoring the entire function might help, I think we start having requirement creep here